cdc/processor/sinkmanager/table_sink_wrapper.go

```diff
@@ -417,6 +419,31 @@ func (t *tableSinkWrapper) cleanRangeEventCounts(upperBound engine.Position, min
 	return shouldClean
 }
 
+func (t *tableSinkWrapper) sinkMaybeStuck(stuckCheck time.Duration) (bool, uint64) {
+	t.getCheckpointTs()
+
+	t.tableSink.RLock()
+	t.tableSink.innerMu.Lock()
+	version := t.tableSink.version
+	resolvedTs := t.tableSink.resolvedTs
+	checkpointTs := t.tableSink.checkpointTs
+	advanced := t.tableSink.advanced
+	t.tableSink.RUnlock()
+	t.tableSink.innerMu.Unlock()
+
+	// What these conditions mean:
+	// 1. the table sink has been associated with a valid sink;
+	// 2. its checkpoint hasn't been advanced for a while;
+	// 3. but the table upperbound is advanced correctly;
+	// 4. its checkpoint is less than the emitted point.
+	if version > 0 && time.Since(advanced) > stuckCheck &&
+		oracle.GetTimeFromTS(t.getUpperBoundTs()).Sub(oracle.GetTimeFromTS(checkpointTs.Ts)) > stuckCheck &&
+		checkpointTs.Less(resolvedTs) {
```

Comment:
I think it is unnecessary for this condition since the line 222 already assure it